### PR TITLE
Implement setDnsEntries(domain_name, dns_entries) API Call

### PR DESCRIPTION
**Pull request by arjan-s**
This branch implements the getDnsEntries API call of the TransIP API and adds a test for it.

### CODE REVIEW

**Comment by mhogerheijde** [comment]
Cool!

I propose to remove the pylint rule that data-transfer-objects should have public methods, agreed?

File: transip/service/dns.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+"""
+DNS related objects
```
Comment (mhogerheijde) [inline transip/service/dns.py]: Nitpick: Not very descriptive of what this class is for.

File: transip/service/dns.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+"""
+
+from suds.sudsobject import Object as SudsObject
+
+
+class DnsEntry(SudsObject):
+    """
+    Representation of the DnsEntry object
```
Comment (mhogerheijde) [inline transip/service/dns.py]: Self-implied comment... This is _per definition_ the representation of the `DnsEntry` class.

Proposed:

    Representation of a DNS entry which could also be found in the control-panel for TransIP 
    and/or in DNS zone-files, etc..

File: transip/service/dns.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    """
+    TYPE_A = 'A'
+    TYPE_AAAA = 'AAAA'
+    TYPE_CNAME = 'CNAME'
+    TYPE_MX = 'MX'
+    TYPE_NS = 'NS'
+    TYPE_TXT = 'TXT'
+    TYPE_SRV = 'SRV'
```
Comment (mhogerheijde) [inline transip/service/dns.py]: I guess these are all the types that TransIP supports? Is there a way to retrieve this list from their API?

File: transip/service/dns.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    name = None
+    expire = 0
+    type = None
+    content = None
+
+    def __init__(self, name, expire, type, content):
```
Comment (mhogerheijde) [inline transip/service/dns.py]: the parameter `type` will mask the [build in function](https://docs.python.org/2/library/functions.html#type)

proposed: `entry_type` or `record_type`

File: tests/service_tests/test_domain.py [hunk 5]
```diff
@@ -52,3 +52,23 @@ def testGetInfo(self, mock_client):
...
+        ds.build_cookie = Mock(return_value={"cookie": "value"})
+        ds.update_cookie = Mock()
+
+        i = mock_client.return_value
+        i.service.setDnsEntries.return_value = None
+
+        dns_entry = DnsEntry('testentry', 86400, DnsEntry.TYPE_A, '127.0.0.1')
+
+        # CALL
+        result = ds.set_dns_entries('domain1', [dns_entry, ])
```
Comment (mhogerheijde) [inline tests/service_tests/test_domain.py]: what happens when the list is empty and/or contains `None`-types?

File: transip/service/dns.py (hunk 3 above)
Comment (arjan-s) [inline transip/service/dns.py]: Actually, I simply copied these from the PHP API that can be downloaded from the TransIP website. Also, these are the only choices you have on their ControlPanel.

File: transip/service/dns.py (hunk 4 above)
Comment (arjan-s) [inline transip/service/dns.py]: Agreed. Modified it to record_type.

File: transip/service/dns.py (hunk 1 above)
Comment (arjan-s) [inline transip/service/dns.py]: It's the docstring for the whole module, which includes DNS related objects. Not sure what else to put there.

**Comment by arjan-s** [comment]
Agreed on the pylint requirement. Classes like DnsEntry don't need a lot of methods.

File: transip/service/dns.py (hunk 2 above)
Comment (arjan-s) [inline transip/service/dns.py]: Shall I modify it to "Representation of a DNS record as expected by the API" then?

File: tests/service_tests/test_domain.py (hunk 5 above)
Comment (arjan-s) [inline tests/service_tests/test_domain.py]: In both cases the server replies with 'Invalid API signature, signature does not match the request.', which is a suds.WebFault and can be catched.

File: transip/service/dns.py (hunk 1 above)
Comment (mhogerheijde) [inline transip/service/dns.py]: Maybe nothing and remove the pylint restriction? The pylint config is not the law in my opinion, but a safeguard. If it means adding useless comments, I'd rather remove the restriction.

Comment (arjan-s) [inline transip/service/dns.py]: Agreed. But how does this work with pylint? I don't know...

File: transip/service/dns.py (hunk 3 above)
Comment (mhogerheijde) [inline transip/service/dns.py]: Yeah, I think these might be the ones defined by the core RFC, there are a lot of addenda to the core RFC for DNS. Lets leave as-is.

File: tests/service_tests/test_domain.py (hunk 5 above)
Comment (mhogerheijde) [inline tests/service_tests/test_domain.py]: ack.

File: transip/service/dns.py (hunk 2 above)
Comment (mhogerheijde) [inline transip/service/dns.py]: Since this is a simple class object, let's leave the comment out and change the pylint directive.

File: transip/service/dns.py (hunk 1 above)
Comment (mhogerheijde) [inline transip/service/dns.py]: There is a multitude of options ;-) But the best is probably to change the `.pylintrc` file. (See http://docs.pylint.org/ for more info)

The name of the rule that is violated is in braces on the end of the warning/error. e.g. `too-few-public-methods` for `R:  4, 0: Too few public methods (0/2) (too-few-public-methods)`. It's a bit annoying, but rules have names _and_ identifiers. Identifieres for rule-names can be found here: http://docs.pylint.org/features.html. (In this case it's `R0903`)

This rule has a threshold of minimum 2 public methods. It doesn't make sense to lower that to 1, so disabling this rule is probably the best way to go.

Change `.pylintrc` in the `[MESSAGES CONTROL]` section to add `R0903` to the `disable=` property.

Idem for the other error.

Comment (mhogerheijde) [inline transip/service/dns.py]: P.S. You can run
```
pylint --rcfile=.pylintrc transip
```
to test the configuration locally.

P.P.S. I will add this to the docs :-)

Comment (arjan-s) [inline transip/service/dns.py]: Disabling R0903 is alright, but I don't think you want to disable C0111 globally.
As de docs say: `Missing %s docstring Used when a module, function, class or method has no docstring. Some special methods like __init__ doesn’t necessary require a docstring.`
So the check would be disabled not only for modules, but also for classes and methods.
It would be better to just disable it for one module, but I don't think that's possible.
So, long story short: I think it's better to include a short generic docstring than to disable the check for them globally.

Comment (mhogerheijde) [inline transip/service/dns.py]: You can disable checks for a specific file as well, add
```
# pylint: disable=C0321
```
to the file. (Can be done locally as well, by only including it in a block)

I just read that starting from Pylint v. 0.25.3, you can use the symbolic names for messages instead of remembering all those code numbers. E.g.:

```
# pylint: disable=missing-docstring
```

Current version is 1.4, so that would make things a bit more readable.

**Comment by arjan-s** [comment]
What do you think about this version? I disabled too-few-public-methods locally, and added a more descriptive module docstring.

**Comment by mhogerheijde** [comment]
Nice! Thanks :-)
